### ISSUE TITLE
get_observables() encounters error at large ensemble sizes

### ISSUE THREAD

**Issue by JunmiHu** [bgroenks96/SimulationBasedInference.jl#9]
### Description
Hi, 
I encountered an error with the get_observables() function at large ensemble sizes for the ensemble methods. Running get_observables(inference_sol) produces a DimensionMismatch at large ensemble sizes, but works perfectly fine with smaller ensemble sizes. 

The error seems to be produced as a result of mismatching dimensions during some concatenation happening in SimulationBasedInference. 

### To reproduce
I've attached the relevant code from the linearode example below>
```
using SimulationBasedInference
using OrdinaryDiffEq
import Random

const rng = Random.MersenneTwister(1234);

ode_func(u,p,t) = -p[1]*u;
α_true = 0.2
ode_p = [α_true];
tspan = (0.0,10.0);
odeprob = ODEProblem(ode_func, [1.0], tspan, ode_p)

dt = 0.2
tsave = tspan[begin]+dt:dt:tspan[end];
n_obs = length(tsave);
observable = ODEObservable(:y, odeprob, tsave, samplerate=0.01);
forward_prob = SimulatorForwardProblem(odeprob, observable)
ode_solver = Tsit5()
forward_sol = solve(forward_prob, ode_solver);
true_obs = get_observable(forward_sol, :y)
noise_scale = 0.05
noisy_obs = true_obs .+ noise_scale*randn(rng, n_obs);
model_prior = prior(α=Beta(2,2));
noise_scale_prior = prior(σ=Exponential(0.1));

lik = IsotropicGaussianLikelihood(observable, noisy_obs, noise_scale_prior);

inference_prob = SimulatorInferenceProblem(forward_prob, ode_solver, model_prior, lik);
enis_sol = solve(inference_prob, EnIS(), ensemble_size=2048, rng=rng);
get_observables(enis_sol)
```

### Stack trace
```
┌ Warning: `cat` cannot concatenate `Dimension`s, falling back to `parent` type:
│ `Ordered` lookups are misaligned at 1024 and 1 on dimension Dim{:ens}. 
│ 
│ To fix for `AbstractDimArray`, pass new lookup values as `cat(As...; dims=Dim{:ens}(newlookupvals))` keyword or `dims=Dim{:ens}()` for empty `NoLookup`.
└ @ DimensionalData ~/.julia/packages/DimensionalData/BZbYQ/src/array/methods.jl:525
ERROR: DimensionMismatch: Array A has 3 axes, while the number of dims is 2: (Ti, Dim{:ens})
Stacktrace:
  [1] format(dims::Tuple{Ti{…}, Dim{…}}, A::Array{Float64, 3})
    @ DimensionalData.Dimensions ~/.julia/packages/DimensionalData/BZbYQ/src/Dimensions/format.jl:36
  [2] _cat(catdims::Tuple{…}, A1::DimensionalData.DimMatrix{…}, As::DimArray{…})
    @ DimensionalData ~/.julia/packages/DimensionalData/BZbYQ/src/array/methods.jl:327
  [3] _cat
    @ ~/.julia/packages/DimensionalData/BZbYQ/src/array/methods.jl:260 [inlined]
  [4] #cat#196
    @ ~/.julia/packages/DimensionalData/BZbYQ/src/array/methods.jl:257 [inlined]
  [5] enscat(acc::DimensionalData.DimMatrix{…}, x::DimensionalData.DimMatrix{…})
    @ SimulationBasedInference ~/sommerjobb2024/SimulationBasedInference.jl/src/ensembles/ensemble_utils.jl:94
  [6] (::SimulationBasedInference.var"#4#6"{…})(k::Symbol)
    @ SimulationBasedInference ~/sommerjobb2024/SimulationBasedInference.jl/src/utils.jl:17
  [7] map
    @ ./tuple.jl:291 [inlined]
  [8] (::SimulationBasedInference.var"#3#5"{…})(acc::@NamedTuple{…}, xᵢ::@NamedTuple{…})
    @ SimulationBasedInference ~/sommerjobb2024/SimulationBasedInference.jl/src/utils.jl:17
  [9] mapreduce_impl(f::typeof(identity), op::SimulationBasedInference.var"#3#5"{…}, A::Vector{…}, ifirst::Int64, ilast::Int64, blksize::Int64)
    @ Base ./reduce.jl:273
 [10] mapreduce_impl
    @ ./reduce.jl:277 [inlined]
 [11] _mapreduce(f::typeof(identity), op::SimulationBasedInference.var"#3#5"{…}, ::IndexLinear, A::Vector{…})
    @ Base ./reduce.jl:447
 [12] _mapreduce_dim
    @ ./reducedim.jl:365 [inlined]
 [13] mapreduce
    @ ./reducedim.jl:357 [inlined]
 [14] reduce
    @ ./reducedim.jl:406 [inlined]
 [15] ntreduce
    @ ~/SimulationBasedInference.jl/src/utils.jl:16 [inlined]
 [16] get_observables(sol::SimulatorInferenceSolution{…}, iter::Int64)
    @ SimulationBasedInference ~/SimulationBasedInference.jl/src/ensembles/ensemble_utils.jl:82
 [17] top-level scope
    @ ~/SimulationBasedInference.jl/examples/linearode/linearode.jl:31


```

Let me know if you require any more information.

**Comment by bgroenks96** [comment]
Hi,

Thanks for the bug report.

That's very strange.... it looks like it's due to some kind of optimization in the base implementation of `reduce` which batches the operation over some threshold.

Hopefully shouldn't be too hard to fix... probably just need to rethink how `ntreduce` is written in light of this.

**Comment by bgroenks96** [comment]
Ok, yep, easy fix. Just needed to use `foldl` instead of `reduce`, since the former guarantees associativity. That was my mistake for not reading the docstring of `reduce` carefully enough.

Fixed by 97f1ed8

**Comment by JunmiHu** [comment]
Thank you!
